src/search/merge_and_shrink/transition_system.cc

```diff
@@ -245,32 +234,41 @@ void TransitionSystem::compute_locally_equivalent_labels() {
     /*
       Compare every group of labels and their transitions to all others and
       merge two groups whenever the transitions are the same.
+
+      Note that there can be empty local label groups after applying label
+      reduction when combining labels which are combinable for this transition
+      system.
```

Comment:
This looks like an inefficient implementation, but so does the old one. (More of a comment for the future.) Is this sometimes a bottleneck?

Reply:
I might be and the meta issue (https://issues.fast-downward.org/issue567) for refactoring M&S is aware of this. Item C reads:
C. TransitionSystem::compute_locally_equivalent_labels(): make computation
faster through sorting transitions (O(n log n)) or hashing (O(n)). Sorting is
probably better, because we do not need to consider all transitions in the best
case. [This doesn't seem to be a huuuuuge bottleneck, though. According to
experiments (not on the tracker), it can be up to 10% of merge-and-shrink
construction time in some configs.]

Reply:
OK, up to 10% isn't necessarily a deal (although also I wouldn't necessarily trust old numbers like these). In any case, doesn't appear relevant now.